In [25]:
import pandas as pd
import spotipy
from spotipy.oauth2 import SpotifyClientCredentials
import json
import matplotlib.pyplot as plt
import seaborn as sns
sns.set()

In [ ]:
client_credentials_manager = spotipy.oauth2.SpotifyClientCredentials(client_id, client_secret)
spotify = spotipy.Spotify(client_credentials_manager=client_credentials_manager)

# Get TrySail information

In [26]:
# search artist and song id
search = spotify.search("TrySail", limit=50, offset=0, type='track', market=None)
trysail = pd.DataFrame.from_dict(search["tracks"]["items"])
trysail_name = pd.DataFrame(trysail["name"])
trysail = pd.DataFrame(trysail["id"])

# get song information
trysail_info = pd.DataFrame()
for url in song_trysail["id"]:
    df = pd.DataFrame.from_dict(spotify.audio_features(url))
    trysail_info = trysail_info.append(df)

# drop unnecessary information 
trysail_info = trysail_info.reset_index(drop=True)
trysail_info = trysail_info.drop(columns=["analysis_url","id","track_href","type","uri"])

# name concat
trysail_info_all = pd.concat([trysail_name, trysail_info], axis=1)
trysail_info_all.head()

name  acousticness  danceability  duration_ms  energy  \
0            Adrenaline!!!       0.12500         0.603       274147   0.959   
1        High Free Spirits       0.04570         0.605       284200   0.936   
2                    Azure       0.15800         0.605       266107   0.927   
3                   Senpai       0.24900         0.526       312733   0.906   
4                 Original       0.00685         0.499       298533   0.894   
5                 Kakawari       0.01910         0.639       265387   0.908   
6                   Truth.       0.01450         0.511       253227   0.905   
7                BraveSail       0.22100         0.639       306267   0.807   
8                Tail Wind       0.06660         0.572       241627   0.973   
9                   Coding       0.02520         0.544       233213   0.953   
10             Wanted Girl       0.06440         0.672       251480   0.978   
11               Try Again       0.03930         0.538       229933   0.973   
12                Chip Log       0.16600         0.649       261800   0.933   
13                   Disco       0.07730         0.676       227320   0.928   
14  Bang! Bang!! Banzai!!!       0.04920         0.475       289747   0.940   
15          Make Me Happy?       0.21200         0.502       285160   0.890   
16       Bokurano Symphony       0.17600         0.690       221987   0.936   
17     Take a Step Forward       0.06850         0.563       267107   0.924   
18          Mirai Curation       0.02130         0.413       241627   0.975   
19              Sampomichi       0.02780         0.697       239400   0.933   

    instrumentalness  key  liveness  loudness  mode  speechiness    tempo  \
0           0.000016    7    0.3010    -2.003     1       0.0567  133.932   
1           0.000000    8    0.2840    -2.112     1       0.0657   93.070   
2           0.000000    4    0.3280    -3.691     1       0.0311  140.005   
3           0.000000    0    0.0339    -3.843     1       0.0574  161.969   
4           0.000196    8    0.1380    -4.294     1       0.1040  164.098   
5           0.000000    7    0.1100    -2.957     1       0.0286   96.484   
6           0.000000    5    0.8000    -3.945     1       0.0643  174.040   
7           0.000010    9    0.2670    -4.433     1       0.0263  109.976   
8           0.000002    7    0.1430    -2.348     1       0.0639  165.005   
9           0.000023    2    0.3380    -2.992     0       0.0630  159.971   
10          0.000000    9    0.3360    -2.289     1       0.1170  120.070   
11          0.000046    1    0.3770    -1.995     1       0.0637  162.941   
12          0.000027    6    0.3770    -2.227     1       0.0333   77.529   
13          0.000047    0    0.1860    -3.175     1       0.0497  125.037   
14          0.000000    3    0.2690    -3.567     1       0.0658  178.017   
15          0.001780    1    0.0949    -3.735     1       0.0370  202.930   
16          0.000000    4    0.0606    -3.194     1       0.0371  115.967   
17          0.000000    2    0.4500    -1.739     1       0.0611   95.957   
18          0.000019    8    0.2840    -2.529     1       0.1070  175.079   
19          0.000044    0    0.1370    -2.984     0       0.0580  114.978   

    time_signature  valence  
0                4    0.885  
1                4    0.666  
2                4    0.783  
3                4    0.364  
4                3    0.566  
5                4    0.877  
6                4    0.349  
7                4    0.921  
8                4    0.698  
9                4    0.760  
10               4    0.713  
11               4    0.726  
12               4    0.633  
13               4    0.869  
14               4    0.694  
15               4    0.790  
16               4    0.868  
17               4    0.432  
18               4    0.655  
19               4    0.827

In [27]:
trysail_info.head()

acousticness  danceability  duration_ms  energy  instrumentalness  key  \
0       0.12500         0.603       274147   0.959          0.000016    7   
1       0.04570         0.605       284200   0.936          0.000000    8   
2       0.15800         0.605       266107   0.927          0.000000    4   
3       0.24900         0.526       312733   0.906          0.000000    0   
4       0.00685         0.499       298533   0.894          0.000196    8   

   liveness  loudness  mode  speechiness    tempo  time_signature  valence  
0    0.3010    -2.003     1       0.0567  133.932               4    0.885  
1    0.2840    -2.112     1       0.0657   93.070               4    0.666  
2    0.3280    -3.691     1       0.0311  140.005               4    0.783  
3    0.0339    -3.843     1       0.0574  161.969               4    0.364  
4    0.1380    -4.294     1       0.1040  164.098               3    0.566

# Analysis

In [28]:
trysail_info.describe()

acousticness  danceability    duration_ms     energy  instrumentalness  \
count     25.000000     25.000000      25.000000  25.000000         25.000000   
mean       0.137453      0.575120  263922.360000   0.882960          0.036809   
std        0.206337      0.076464   28703.794373   0.145753          0.183582   
min        0.006850      0.413000  205053.000000   0.303000          0.000000   
25%        0.025200      0.511000  241627.000000   0.894000          0.000000   
50%        0.066600      0.571000  266107.000000   0.928000          0.000002   
75%        0.166000      0.639000  285160.000000   0.947000          0.000027   
max        0.962000      0.697000  312733.000000   0.978000          0.918000   

             key   liveness   loudness       mode  speechiness       tempo  \
count  25.000000  25.000000  25.000000  25.000000    25.000000   25.000000   
mean    4.520000   0.260444  -3.483880   0.840000     0.056076  138.197000   
std     3.097311   0.164229   2.116822   0.374166     0.024990   32.630607   
min     0.000000   0.033900 -12.711000   0.000000     0.026300   77.529000   
25%     2.000000   0.137000  -3.843000   1.000000     0.033300  114.978000   
50%     4.000000   0.269000  -2.992000   1.000000     0.057400  140.005000   
75%     7.000000   0.338000  -2.289000   1.000000     0.064300  162.941000   
max     9.000000   0.800000  -1.739000   1.000000     0.117000  202.930000   

       time_signature   valence  
count       25.000000  25.00000  
mean         3.920000   0.67232  
std          0.276887   0.17320  
min          3.000000   0.34900  
25%          4.000000   0.56600  
50%          4.000000   0.69800  
75%          4.000000   0.79000  
max          4.000000   0.92100

- acousticness: A confidence measure from 0.0 to 1.0 of whether the track is acoustic. 
- danceability: Degree of fitness for dance (0.0 to 1.0)
- duration_ms: Song length.
- energy: Energetic tracks feel fast, loud, and noisy (0.0 to 1.0)
- instrumentalness: The closer the instrumentalness value is to 1.0, the greater likelihood the track contains no vocal content. Values above 0.5 are intended to represent instrumental tracks, but confidence is higher as the value approaches 1.0.
- key: The key the track is in. Integers map to pitches using standard Pitch Class notation
- liveness: Detects the presence of an audience in the recording. Higher liveness values represent an increased probability that the track was performed live.
- loudness: The overall loudness of a track in decibels (dB). Values typical range between -60 and 0 db.
- mode: Mode indicates the modality (major or minor) of a track, the type of scale from which its melodic content is derived. Major is represented by 1 and minor is 0.
- speechiness: data > 0.66, made of spoken. 0.33 < data < 0.66, both talk and music including rap music. data < 0.33, music.
- tempo: BPM
- time_signature: An estimated overall time signature of a track. 
- valence: A measure from 0.0 to 1.0 describing the musical positiveness conveyed by a track. Tracks with high valence sound more positive (e.g. happy, cheerful, euphoric), while tracks with low valence sound more negative (e.g. sad, depressed, angry).

# Reference

- https://developer.spotify.com/documentation/web-api/reference/tracks/get-several-audio-features/
- https://qiita.com/kazuya-n/items/fbee07ef778e166cb6dd
- https://note.mu/hkrrr_jp/n/n9f33734d2cd6